baseplate/frameworks/queue_consumer/kombu.py

```diff
@@ -161,17 +287,18 @@ def handle(self, message: kombu.Message) -> None:
                 self.handler_fn(context, message_body, message)
         except Exception as exc:
             prometheus_success = "false"
-            logger.exception(
-                "Unhandled error while trying to process a message.  The message "
-                "has been returned to the queue broker."
-            )
+
+            # Custom error_handler_fn has priority over standard handler.
             if self.error_handler_fn:
+                logger.debug(
+                    "Unhandled error while trying to process a message. Custom handler invoked."
+                )
                 self.error_handler_fn(context, message_body, message, exc)
             else:
-                message.requeue()
+                self._handle_error(message, prometheus_labels, exc)
```

Comment:
This function only gets called if the error is not associated with the message handle function. Is this intended?

Reply:
Yes, it's intended, and it's a pre-existing behavior:
```
            if self.error_handler_fn:
                self.error_handler_fn(context, message_body, message, exc)
            else:
                message.requeue()
```
I.e. behavior, as described in the design doc, is extended. Before this change users had 2 options: to write their custom error handler, or to rely on default strategy of infinite retries. Now they have 3 options: custom error handler, configurable republishing mechanism (that can support retries limiting) or the old default.